src/server.mjs

```diff
@@ -1094,17 +1118,148 @@ api.post("/getPathNames", (req, res, next) => {
   });
 });
 
-api.post("/getBedRegions", (req, res) => {
+// returns whether or not a string is a valid http url
+function isValidURL(string) {
+  if (!string) {
+    return False;
+  }
+
+  let url;
+
+  try {
+    url = new URL(string)
+  } catch(error) {
+    return false;
+  }
+
+  return url.protocol === "http:" || url.protocol === "https:";
+}
+
+// given the datapath and url, download file to the datapath
+const downloadFile = async(fileURL, destinationPath, fileName) => {
+
+  if (!fileName) {
+    return;
+  }
+
+  const response = await fetchAndValidate(fileURL, config.maxFileSizeBytes);
+  
+  const destinationFile = path.resolve(destinationPath, fileName);
+
+  // overwrites file if it already exists
+  const fileStream = fs.createWriteStream(destinationFile, { flags: 'w' });
+  await finished(Readable.fromWeb(response.body).pipe(fileStream));
+
+}
+
+const fetchAndValidate = async(url, maxBytes) => {
+  const options = {
+    method: "GET",
+    credentials: "omit",
+    cache: "default",
+    signal: timeoutController(config.fetchTimeout).signal
+  };
+
+  let response = await fetch(url, options);
+
+  // check for unsuccessful response codes
+  if (!response.ok) {
+    throw new BadRequestError("Fetch request failed: " + response.status);
+  }
+
+  // check for size specified in header
+  const contentLength = response.headers.get("Content-Length");
+
+  if (contentLength > maxBytes) {
+    throw new Error("Fetch request failed: contentLength exceeds maximum file size");
+  }
+
+  // use a reader to make sure we're not reading past the max size allowed
+  const reader = response.body.getReader();
+
+  let bytesRead = 0;
+  const dataRead = [];
+
+  while (true) {
+    let {done, value} = await reader.read();
+
+    if (done) {
+      break;
+    }
+
+    dataRead.push(value);
+    bytesRead += new Blob([value]).size;
+
+    if (bytesRead > maxBytes) {
+      reader.cancel();
+      throw new Error("Fetched request failed, maximum file size exceeded: " + maxSize);
+    }
+  }
+
+  return new Response(new Blob(dataRead), { headers: response.headers });
+
+}
+
+// downloads files fo the specified chunk name from the destination of the bedURL
+// includeContent only downloads the tracks.json file when set to false
```

Comment:
I don't think this is a good design. When used with a false flag, this isn't *really* retrieving the chunk in a meaningful sense.

If we actually do want to create N directories, each with a base URL file and a track metadata file, when we first see a BED file with N entries, then we should split that out from downloading the actual chunk files. We don't need the code to make the base URL file when actually retrieving the full chunk, and we don't need the code to look at the `chunk_contents.txt` file when retrieving `tracks.json` (because we already know the name and we can just fetch it if it exists).